core/templates/dev/head/editor/feedback_tab/ThreadDataService.js

```diff
@@ -114,6 +126,14 @@ oppia.factory('threadDataService', [
       };
 
       $http.post(url, payload).success(function() {
+        if (updatedStatus != null) {
```

Comment:
I suggest moving most/all of this to where you deleted "thread.status = newStatus". It's a common pattern to perform UI actions as though the $http call has succeeded, so that the user does not experience a delay due to the server lag. In the rare case where the call fails, an error should be shown (and the changes should get undone or a page reload should be forced -- we should probably do the former, in this case).

Another reason for doing this is that the code here makes assumptions on what happened before the $http.post (e.g. that updatedStatus is always different from thread.status). So I lean either towards keeping all that logic in one place, or else making the checks here stronger and more robust (if old thread status is X and new thread status is Y then do Z), to prevent cases in the future where a developer accidentally changes one part of the code but forgets to change the other.